File: golang/_0011_Container_With_Most_Water/main.go
package main

func maxArea(height []int) int {
	n, res := len(height), 0
	if n >= 2 {
		left, right := 0, n-1

		for left < right {
			h := 0
			if height[left] < height[right] {
				h = height[left]
				left++
			} else {
				h = height[right]
				right--
			}

			res = max(res, h*(right-left+1))
		}
	}

	return res
}

func max(i, j int) int {
	if i > j {
		return i
	} else {
		return j
	}
}

File: python3/_0011_Container_With_Most_Water/main.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        res, n = 0, len(height)
        if n > 1:
            left, right = 0, n - 1
            while left < right:
                res = max(res, (right - left) * min(height[left], height[right]))
                if height[left] <= height[right]:
                    left += 1
                else:
                    right -= 1

        return res
